doc/source/rllib-training.rst

```diff
@@ -520,6 +520,142 @@ Custom metrics can be accessed and visualized like any other training result:
+        exploratory action.
+
+        Args:
+            distribution_inputs (any): The output coming from the model,
+                ready for parameterizing a distribution
+                (e.g. q-values or PG-logits).
+            action_dist_class (class): The action distribution class
+                to use.
+            model (ModelV2): The Model object.
+            explore (bool): True: "Normal" exploration behavior.
+                False: Suppress all exploratory behavior and return
+                    a deterministic action.
+            timestep (int): The current sampling time step. If None, the
+                component should try to use an internal counter, which it
+                then increments by 1. If provided, will set the internal
+                counter to the given value.
+
+        Returns:
+            Tuple:
+            - The chosen exploration action or a tf-op to fetch the exploration
+              action from the graph.
+            - The log-likelihood of the exploration action.
+        """
+        pass
+
+It takes the model’s output, the action distribution class, the model itself,
+a timestep, and an ``explore`` switch and outputs a tuple of 1) action and
+2) log-likelihood.
+
+On the highest level, the ``Trainer.compute_action`` and ``Policy.compute_action(s)``
+methods have a boolean ``explore`` switch, which is passed into
+``Exploration.get_exploration_action``. If ``None``, the value of
+``Trainer.config[“explore”]`` is used.
+Hence ``config[“explore”]`` describes the default behavior of the policy and
+e.g. allows switching off any exploration easily for evaluation purposes
+(see :ref:`CustomEvaluation`).
+
+The following are example excerpts from different Trainers' configs
+(see rllib/agents/trainer.py) to setup different exploration behaviors:
+
+.. code-block:: python
+
+    # All of the following configs go into Trainer.config.
+
+    # 1) Switching *off* exploration by default.
+    # Behavior: Calling `compute_action(s)` without explicitly setting its `explore`
+    # param will result in no exploration.
+    # However, explicitly calling `compute_action(s)` with `explore=True` will
+    # still(!) result in exploration (per-call overrides default).
+    "explore": False,
+
+    # 2) Switching *on* exploration by default.
+    # Behavior: Calling `compute_action(s)` without explicitly setting its
+    # explore param will result in exploration.
+    # However, explicitly calling `compute_action(s)` with `explore=False`
+    # will result in no(!) exploration (per-call overrides default).
+    "explore": True,
+
+    # 3) Example exploration_config usages:
+    # a) DQN: see rllib/agents/dqn/dqn.py
+    "explore": True,
+    "exploration_config": {
+       "type": "EpsilonGreedy",  # <- Exploration sub-class by name or full path to module+class
+                                 # (e.g. “ray.rllib.utils.exploration.epsilon_greedy.EpsilonGreedy”)
```

Comment:
This makes the code block overflow horizontally on my machine, consider trimming it down by a few words.

Reply:
+1

Reply:
Fixed.